### PR TITLE
Adding NMState operator and e2e

### PR DESCRIPTION
**Pull request by varad-ahirwadkar**
Signed-off-by: Varad Ahirwadkar <varad.ahirwadkar@ibm.com>

### CODE REVIEW

File: playbooks/roles/ocp-nmstate-operator/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      name: openshift-nmstate
+
+  - name: Check if the namespace gets deleted
+    shell: oc get namespace --no-headers | grep nmstate | wc -l
+    register: nmstate_namespace
+    until: nmstate_namespace.stdout|int == 0
+    retries: 15
+    delay: 30
+
+  - name: Delete the workplace
```
Comment (pravin-dsilva) [inline playbooks/roles/ocp-nmstate-operator/tasks/main.yml:99]:
```suggestion
  - name: Delete the workspace
```

**Review by pravin-dsilva — CHANGES_REQUESTED** [review]

File: examples/nmstate_vars.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+nmstate_enabled: false
+nmstate_install_operator: true
+nmstate_catalogsource_image: "" 
+nmstate_upgrade_channel: "stable"
+nmstate_directory: "/tmp/nmstate"
+nmstate_golang_tarball: "https://go.dev/dl/go1.18.6.linux-ppc64le.tar.gz"
+nmstate_e2e: false
+nmstate_e2e_git_repository: ""
```
Comment (yussufsh) [inline examples/nmstate_vars.yaml:9]: I know this is not needed when e2e flag is false, but can we provide example values here? Empty string in examples does not make sense for people referring it to it. It is same as default variable files.

Comment (varad-ahirwadkar) [reply examples/nmstate_vars.yaml]: I have added example

Comment (yussufsh) [reply examples/nmstate_vars.yaml]: @varad-ahirwadkar This is an example file, you do not need to comment with "eg.". Directly specify the value.

**Review by yussufsh — APPROVED** [review]

**Review by pravin-dsilva — APPROVED** [review]

**Pull request merged**